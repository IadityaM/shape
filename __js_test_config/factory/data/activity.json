{"data":{"id":"1365201","type":"activities","attributes":{"action":null,"created_at":"2019-12-16T23:14:35.973Z","content":null,"actor_id":22548,"target_type":"Collection","target_id":762911,"source_name":""},"relationships":{"actor":{"meta":{"included":false}},"subject_users":{"meta":{"included":false}},"subject_groups":{"meta":{"included":false}},"target":{"meta":{"included":false}}}}}