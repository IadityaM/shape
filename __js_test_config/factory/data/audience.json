{"data":{"id":"104","type":"audiences","attributes":{"name":"Natural Armor","global_default":null,"age_list":[],"children_age_list":[],"country_list":[],"education_level_list":[],"gender_list":[],"adopter_type_list":[],"interest_list":[],"publication_list":[],"min_price_per_response":4.0,"order":null,"global":true}}}