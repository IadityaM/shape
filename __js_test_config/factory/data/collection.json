{"data":{"id":"762903","type":"collections","attributes":{"class_type":"Collection","name":"Extended","created_at":"2019-12-16T23:14:33.643Z","updated_at":"2019-12-16T23:14:33.643Z","master_template":false,"template_id":null,"submission_box_type":null,"submission_box_id":null,"submission_template_id":null,"test_status":null,"collection_to_test_id":null,"hide_submissions":false,"submissions_enabled":true,"anyone_can_view":false,"anyone_can_join":false,"cover_type":"cover_type_default","archived":false,"unresolved_count":0,"last_unresolved_comment_id":null,"test_show_media":true,"idea_id":null,"search_term":null,"organization_id":"5352","joinable_group_id":null,"default_group_id":null,"system_required":false,"tag_list":[],"inherited_tag_list":[],"cover":{"text":"","hardcoded_subtitle":"","subtitle_hidden":false},"test_scores":{},"breadcrumb":[{"type":"collections","id":"762903","name":"Extended","can_edit":false}],"processing_status":null,"collection_card_count":0,"can_view":false,"can_edit":false,"can_edit_content":false,"is_private":false,"is_org_template_collection":false,"is_profile_template":false,"is_profile_collection":false,"pinned_and_locked":false,"is_submission_box_template":false,"is_submission_box_template_test":false,"is_inside_hidden_submission_box":false,"is_inside_a_submission":false,"is_subtemplate_or_instance":false,"show_language_selector":false,"template_num_instances":0,"awaiting_updates":null,"num_survey_responses":0,"max_row_index":null,"max_col_index":null,"frontend_url":"http://localhost/","common_viewable":null,"is_test_locked":null,"has_link_sharing":null,"is_restorable":false,"serializer":"SerializableCollection"},"relationships":{"roles":{"meta":{"included":false}},"collection_cover_items":{"meta":{"included":false}},"parent_collection_card":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"live_test_collection":{"meta":{"included":false}},"submissions_collection":{"meta":{"included":false}},"submission_template":{"meta":{"included":false}},"collection_to_test":{"meta":{"included":false}},"organization":{"meta":{"included":false}},"created_by":{"meta":{"included":false}},"test_audiences":{"meta":{"included":false}},"test_results_collection":{"meta":{"included":false}},"collection_filters":{"meta":{"included":false}},"restorable_parent":{"meta":{"included":false}}}}}