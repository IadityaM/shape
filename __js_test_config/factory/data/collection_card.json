{"data":{"id":"2877260","type":"collection_cards","attributes":{"class_type":"CollectionCard::Primary","order":0,"width":1,"height":1,"parent_id":762904,"pinned":false,"image_contain":false,"card_question_type":null,"is_cover":false,"filter":"transparent_gray","hidden":false,"show_replace":true,"updated_at":"2019-12-16T23:14:34.478Z","col":1,"row":1,"section_type":null,"idea_id":null,"link":false,"pinned_and_locked":false,"can_edit_parent":false,"is_master_template_card":false},"relationships":{"item":{"meta":{"included":false}},"collection":{"meta":{"included":false}},"parent":{"meta":{"included":false}},"record":{"meta":{"included":false}},"templated_from":{"meta":{"included":false}}}}}