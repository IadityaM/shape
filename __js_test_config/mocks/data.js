export const fakeTextItemAttrs = {
  id: 1,
  type: 'Item::TextItem',
  name: 'My Cool Item',
  text_data: 'This is the content for the item and it contains multiple sentences. Like this one.',
  breadcrumb: [['collections', 1, 'Some collection'], ['items', 1, 'my item']],
  parentPath: '/',
}
export const fakeTextItem = {
  ...fakeTextItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
  toJS: jest.fn().mockReturnValue(fakeTextItemAttrs),
}
export const fakeVideoItemAttrs = {
  id: 2,
  type: 'Item::VideoItem',
  name: 'Crazy Honey Badger',
  url: 'https://www.youtube.com/watch?v=4r7wHMg5Yjg'
}
export const fakeVideoItem = {
  ...fakeVideoItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeImageItemAttrs = {
  id: 3,
  type: 'Item::ImageItem',
  name: 'Earth from Space',
  filestack_file: {
    id: 1,
    url: 'https://www.nasa.gov/sites/default/files/styles/full_width_feature/public/thumbnails/image/iss052e023801_0.jpg',
  }
}
export const fakeImageItem = {
  ...fakeImageItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeItemCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeTextItem,
  item: fakeTextItem,
}
export const fakeCollectionCard = {
  id: 11,
  order: 1,
  height: 1,
  width: 1,
  record: fakeCollection,
  item: fakeCollection,
  reference: false,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  type: 'Collection',
  breadcrumb: [],
  collection_cards: [
    fakeItemCard, fakeItemCard, fakeItemCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
