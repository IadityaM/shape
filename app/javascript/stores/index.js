import { config as jsonApiConfig, Store } from 'mobx-jsonapi-store'
import { RouterStore } from 'mobx-react-router'

import locale from './Locale'
import User from './User'
import Collection from './Collection'
import Item from './Item'
import CollectionCard from './CollectionCard'

jsonApiConfig.baseUrl = '/api/v1/'
// modify fetch to include 'same-origin' credentials
jsonApiConfig.fetchReference = (url, opts) => {
  opts.credentials = 'same-origin'
  return fetch(url, opts)
}

export const routingStore = new RouterStore()

class ApiStore extends Store {}
ApiStore.types = [User, Collection, Item, CollectionCard]

export const apiStore = new ApiStore()

export default {
  routingStore,
  apiStore,
  // needs to be named "locale"
  locale
}
