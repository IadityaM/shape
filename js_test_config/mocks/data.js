export const fakeItemAttrs = {
  id: 1,
  name: 'My Cool Item',
}
export const fakeItem = {
  ...fakeItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeItem,
  item: fakeItem,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  breadcrumb: [],
  collection_cards: [
    fakeCard, fakeCard, fakeCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
