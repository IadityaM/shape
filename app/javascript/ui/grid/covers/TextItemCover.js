import _ from 'lodash'
import PropTypes from 'prop-types'
import ReactDOM from 'react-dom'
import { computed, runInAction } from 'mobx'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import ReactQuill from 'react-quill'
import styled from 'styled-components'

import { apiStore, routingStore, uiStore } from '~/stores'
import v from '~/utils/variables'
import { ShowMoreButton } from '~/ui/global/styled/forms'
import { QuillStyleWrapper } from '~/ui/global/styled/typography'
import InlineLoader from '~/ui/layout/InlineLoader'
import RealtimeTextItem from '~/ui/items/RealtimeTextItem'
import PaddedCardCover from '~/ui/grid/covers/PaddedCardCover'
const stripTags = str => str.replace(/(<([^>]+)>)/gi, '')

const StyledPaddedCover = styled(PaddedCardCover)`
  border-top: ${props =>
    !props.isEditing && props.hasTitleText
      ? `2px solid ${props.theme.fontColor || v.colors.black}`
      : 'none'};
  background: ${props => {
    const { hasTitleText, isTransparent, uneditable } = props
    if (hasTitleText && uneditable) {
      // for carousel covers w/ title text
      return `${v.colors.white}`
    }
    if (isTransparent || hasTitleText) {
      return `${v.colors.transparent}`
    }
    return `${v.colors.white}`
  }};
`

const StyledReadMore = styled(ShowMoreButton)`
  z-index: ${v.zIndex.gridCard};
  position: absolute;
  bottom: 0;
  left: 0;
  padding: 0.5rem;
  opacity: 0.975;
  background: ${props =>
    !props.isEditing && props.hasTitleText
      ? v.colors.transparent
      : v.colors.white};

  &:hover {
    background: ${v.colors.commonLightest};
  }
`
StyledReadMore.displayName = 'StyledReadMore'

@observer
class TextItemCover extends React.Component {
  lastClickTimestamp = null

  constructor(props) {
    super(props)
    this.unmounted = false
  }

  state = {
    // NOTE: item is stored in state, and set only on mount/update to protect from
    //  observable changes triggering problematic re-renders of RealtimeTextItem
    item: null,
    readMore: false,
    loading: false,
  }

  componentDidMount() {
    const { item, height } = this.props
    this.checkTextAreaHeight(height)
    this.setState({ item })
  }

  componentDidUpdate(prevProps) {
    const { item, height } = this.props
    if (height !== prevProps.height) {
      this.checkTextAreaHeight(height)
    }
    if (item.id !== prevProps.item.id) {
      this.setState({ item })
    }
  }

  componentWillUnmount() {
    this.unmounted = true
  }

  @computed
  get isEditing() {
    const { item, cardId } = this.props
    return (
      uiStore.textEditingItem === item && uiStore.textEditingCardId === cardId
    )
  }

  handleClick = e => {
    const now = new Date().getTime()
    const timeDiff = now - this.lastClickTimestamp
    if (timeDiff < 600 && timeDiff > 0) {
      return this.onOpenTextItem(e)
    }
    this.lastClickTimestamp = new Date().getTime()
    if (uiStore.isTouchDevice) {
      const { cardId } = this.props
      uiStore.openTouchActionMenu(cardId)
    } else {
      return this.onOpenTextItem(e)
    }
  }

  onOpenTextItem(e) {
    if (this.props.handleClick) this.props.handleClick(e)
    e.stopPropagation()
    const { item, dragging, cardId, searchResult, uneditable } = this.props
    if (dragging || uiStore.dragging || this.isEditing || uneditable)
      return false
    // allow both editors/viewers to capture keyboard clicks
    if (uiStore.captureKeyboardGridClick(e, cardId)) {
      return false
    }
    if (!item.can_view) {
      uiStore.showPermissionsAlert()
      return false
    } else if (!item.can_edit_content || searchResult) {
      // if a viewer, there's nothing to do on the generic click action
      // likewise on search results, never pop open the inline editor
      return false
    }
    this.setState({ loading: true }, this.loadItem)
    return null
  }

  loadItem = async () => {
    const { item, cardId } = this.props
    await apiStore.fetch('items', item.id, true)
    // entering edit mode should deselect all cards
    runInAction(() => {
      uiStore.deselectCards()
      uiStore.update('textEditingItemHasTitleText', this.hasTitleText)
      uiStore.update('textEditingItem', item)
      uiStore.update('textEditingCardId', cardId)
    })
    this.setState({ loading: false })
  }

  expand = () => {
    const { item } = this.props
    routingStore.routeTo('items', item.id)
  }

  clearTextEditingItem = () => {
    if (!this.isEditing) return
    uiStore.clearTextEditingItem()
  }

  // cancel should only ever be called for editors, since it is canceling out of edit view
  cancel = ({ item, ev, num_viewers = 1 } = {}) => {
    if (this.unmounted) {
      return
    }
    if (ev && ev.stopPropagation) ev.stopPropagation()
    this.clearTextEditingItem()
    const hasContent = stripTags(item.content).length
    if (!hasContent && item.version === 1) {
      // archive empty text item when you hit "X"
      const card = apiStore.find('collection_cards', this.props.cardId)
      if (card) card.API_archiveSelf({ undoable: false })
      return
    }
    if (num_viewers === 1) {
      // save final updates and broadcast to collection
      item.API_updateWithoutSync({ cancel_sync: true })
    }
    // TODO figure out why ref wasn't working
    // eslint-disable-next-line react/no-find-dom-node
    try {
      const node = ReactDOM.findDOMNode(this)
      node.scrollTop = 0
    } catch {
      // console.warn('probably in jest test')
    }
  }

  checkTextAreaHeight = height => {
    if (!this.reactQuillRef) return
    if (this.props.hideReadMore) return
    // The height of the editor is constrained to the container,
    // we must get the .ql-editor div to calculate text height
    const qlEditor = this.reactQuillRef.editingArea.getElementsByClassName(
      'ql-editor'
    )[0]
    const textAreaHeight = qlEditor ? qlEditor.scrollHeight : 0
    // render the Read More link if the text height exceeds viewable area
    if (height && textAreaHeight > height) {
      this.setState({ readMore: true })
    } else {
      this.setState({ readMore: false })
    }
  }

  get quillEditor() {
    const { reactQuillRef } = this
    if (!reactQuillRef) return

    return reactQuillRef.getEditor()
  }

  renderEditing() {
    const { item } = this.state
    const { initialSize, cardId } = this.props
    if (!item) return ''

    return (
      <RealtimeTextItem
        cardId={cardId}
        item={item}
        currentUserId={apiStore.currentUser.id}
        onExpand={item.id ? this.expand : null}
        onCancel={this.cancel}
        initialSize={initialSize}
        // if we are rendering editing then the item has been fetched
        fullyLoaded
      />
    )
  }

  renderDefault() {
    const { item, cardId } = this.props
    const textData = item.toJSON().quill_data
    const quillProps = {
      // ref is used to get the height of the div in checkTextAreaHeight
      ref: c => {
        this.reactQuillRef = c
      },
      readOnly: true,
      onChangeSelection: (range, source, editor) => {
        const { quillEditor } = this
        uiStore.selectTextRangeForCard({
          range,
          quillEditor,
          cardId,
        })
      },
      theme: null,
    }

    return <ReactQuill {...quillProps} value={textData} />
  }

  get hasTitleText() {
    const { isEditing } = this
    const { item } = this.props
    const { quill_data } = item
    if (isEditing) {
      return uiStore.textEditingItemHasTitleText
    }
    let hasTitle = false
    _.each(quill_data.ops, op => {
      if (op.attributes && op.attributes.header === 5) {
        hasTitle = true
      }
    })
    return hasTitle
  }

  render() {
    const { isEditing, hasTitleText, props } = this
    const { isTransparent, uneditable } = props
    const content = isEditing ? this.renderEditing() : this.renderDefault()

    return (
      <StyledPaddedCover
        data-cy="TextItemCover"
        style={{
          // account for padding: 1rem on both sides
          height: 'calc(100% - 2rem)',
        }}
        className="cancelGridClick"
        onClick={this.handleClick}
        hasTitleText={hasTitleText}
        isTransparent={isTransparent}
        uneditable={uneditable}
        isEditing={isEditing}
      >
        <QuillStyleWrapper
          notEditing={!isEditing}
          smallGrid={uiStore.isSmallGrid}
          hasTitleText={hasTitleText}
        >
          {this.state.loading && <InlineLoader />}
          {content}
          {this.state.readMore && !isEditing && (
            <StyledReadMore
              onClick={this.expand}
              isEditing={isEditing}
              hasTitleText={hasTitleText}
            >
              Read more...
            </StyledReadMore>
          )}
        </QuillStyleWrapper>
      </StyledPaddedCover>
    )
  }
}

TextItemCover.propTypes = {
  item: MobxPropTypes.objectOrObservableObject.isRequired,
  dragging: PropTypes.bool.isRequired,
  cardId: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  initialSize: PropTypes.string.isRequired,
  height: PropTypes.number,
  searchResult: PropTypes.bool,
  hideReadMore: PropTypes.bool,
  uneditable: PropTypes.bool,
  isTransparent: PropTypes.bool,
}

TextItemCover.defaultProps = {
  height: null,
  searchResult: false,
  hideReadMore: false,
  uneditable: false,
  isTransparent: false,
  handleClick: null,
}

export default TextItemCover
