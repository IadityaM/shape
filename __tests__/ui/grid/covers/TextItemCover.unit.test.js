import TextItemCover from '~/ui/grid/covers/TextItemCover'
import { apiStore, uiStore } from '~/stores'

import { fakeTextItem } from '#/mocks/data'

jest.mock('../../../../app/javascript/stores/index')

const item = fakeTextItem
const props = {
  item,
  dragging: false,
  height: 200,
  cardId: '1',
  handleClick: jest.fn(),
  initialSize: 'normal',
}
const e = {
  stopPropagation: jest.fn(),
}
const fakeReactQuillRef = {
  editingArea: {
    getElementsByClassName: jest.fn().mockReturnValue([{ scrollHeight: 500 }]),
  },
  getEditor: jest.fn(),
}

let wrapper, component
describe('TextItemCover', () => {
  beforeEach(() => {
    props.editable = false
    wrapper = shallow(<TextItemCover {...props} />)
    component = wrapper.instance()
  })

  it('renders Quill with item.quill_data', () => {
    expect(wrapper.find('Quill').props().value).toBe(item.quill_data)
  })

  it('renders Read More if text height exceeds the viewable area', () => {
    const inst = wrapper.instance()
    inst.reactQuillRef = fakeReactQuillRef
    inst.componentDidMount()
    // force re-render to pick up state update
    wrapper.update()
    expect(wrapper.state().readMore).toBe(true)
    expect(wrapper.find('StyledReadMore').exists()).toBe(true)
  })

  it('does not render Read More if text height fits within the viewable area', () => {
    const inst = wrapper.instance()
    inst.reactQuillRef = {
      ...fakeReactQuillRef,
      editingArea: {
        getElementsByClassName: jest
          .fn()
          .mockReturnValue([{ scrollHeight: 100 }]),
      },
    }
    inst.componentDidMount()
    wrapper.update()
    expect(wrapper.state().readMore).toBe(false)
    expect(wrapper.find('StyledReadMore').exists()).toBe(false)
  })

  describe('handleClick', () => {
    let fakeCard
    beforeEach(() => {
      fakeCard = { id: '1', record: {} }
      apiStore.find = jest.fn().mockReturnValue(fakeCard)
    })

    it('calls props handleClick if it exists', () => {
      component.handleClick(e)
      expect(props.handleClick).toHaveBeenCalled()
    })

    it('returns false if you are dragging', async () => {
      wrapper.setProps({ dragging: true })
      const result = await component.handleClick(e)
      expect(result).toBe(false)
    })

    it("returns false if you can't edit content", async () => {
      wrapper.setProps({
        dragging: false,
        item: { ...item, can_edit_content: false },
      })
      const result = await component.handleClick(e)
      expect(result).toBe(false)
    })

    it('returns false if searchResult is true', async () => {
      wrapper.setProps({
        dragging: false,
        item: { ...item, searchResult: true },
      })
      const result = await component.handleClick(e)
      expect(result).toBe(false)
    })

    it('calls uiStore.setTextEditingCard if can_edit_content', async () => {
      wrapper.setProps({
        dragging: false,
        item: { ...item, can_edit_content: true },
      })
      const result = await component.handleClick(e)
      expect(result).toBe(null)
      expect(uiStore.setTextEditingCard).toHaveBeenCalledWith(fakeCard, {
        hasTitleText: false,
      })
    })

    it('calls apiStore.fetch item if can_edit_content', async () => {
      wrapper.setProps({
        dragging: false,
        item: { ...item, can_edit_content: true },
      })
      const result = await component.handleClick(e)
      expect(result).toBe(null)
      expect(apiStore.fetch).toHaveBeenCalledWith('items', item.id, true)
    })

    it('calls uiStore.showPermissionsAlert if cannot view', () => {
      wrapper.setProps({
        item: { ...item, can_view: false },
      })
      component.handleClick(e)
      expect(uiStore.showPermissionsAlert).toHaveBeenCalled()
    })
  })

  describe('cancel', () => {
    beforeEach(() => {
      item.content = '<p></p>'
      item.persisted = true
      item.API_updateWithoutSync = jest.fn()
    })

    describe('with content', () => {
      beforeEach(() => {
        item.content = '<p>some content</p>'
      })

      describe('with num_viewers === 1', () => {
        describe('with unpersisted item (for quick text creation)', () => {
          it('should not call API_updateWithoutSync', () => {
            item.persisted = false
            component.cancel({ item, ev: e })
            expect(item.API_updateWithoutSync).not.toHaveBeenCalled()
            // should clear out temp items
            expect(uiStore.clearTempTextCardItems).toHaveBeenCalled()
          })
        })

        describe('with item.persisted', () => {
          it('should call API_updateWithoutSync', () => {
            item.persisted = true
            component.cancel({ item, ev: e })
            expect(item.API_updateWithoutSync).toHaveBeenCalled()
          })
        })
      })

      describe('with num_viewers > 1', () => {
        it('should not call API_updateWithoutSync', () => {
          component.cancel({ item, ev: e, num_viewers: 2 })
          expect(item.API_updateWithoutSync).not.toHaveBeenCalled()
        })
      })
    })

    describe('with no content', () => {
      const card = { API_archiveSelf: jest.fn() }
      beforeEach(() => {
        item.content = ''
        item.version = null
        apiStore.find = jest.fn().mockReturnValue(card)
        card.API_archiveSelf.mockClear()
      })
      describe('and version === 1', () => {
        beforeEach(() => {
          item.version = 1
          wrapper = shallow(<TextItemCover {...props} />)
          component = wrapper.instance()
        })
        it('calls card.API_archiveSelf', async () => {
          component.cancel({ item, ev: e })
          expect(apiStore.find).toHaveBeenCalledWith(
            'collection_cards',
            props.cardId
          )
          expect(card.API_archiveSelf).toHaveBeenCalled()
        })
      })
      describe('and a version > 1', () => {
        beforeEach(() => {
          item.version = 2
          wrapper = shallow(<TextItemCover {...props} />)
          component = wrapper.instance()
        })
        it('does not call card.API_archiveSelf', async () => {
          component.cancel({ item, ev: e })
          expect(card.API_archiveSelf).not.toHaveBeenCalled()
        })
      })
    })
  })
})
