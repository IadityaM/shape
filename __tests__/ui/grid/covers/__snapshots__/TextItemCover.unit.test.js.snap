// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextItemCover renders snapshot 1`] = `
<styled.div
  class="cancelGridClick"
  data-cy="TextItemCover"
  onClick={[Function]}
  style={
    Object {
      "height": "calc(100% - 30px)",
      "overflowX": "hidden",
      "overflowY": "hidden",
    }
  }
>
  <styled.div
    notEditing={true}
  >
    <Quill
      modules={Object {}}
      readOnly={true}
      theme={null}
      value="This is the content for the item and it contains multiple sentences. Like this one."
    />
  </styled.div>
</styled.div>
`;
